src/sql/azdata.proposed.d.ts

```diff
@@ -741,6 +745,15 @@ declare module 'azdata' {
 			viewModel: DesignerViewModel;
 		}
 
+		/**
+		 * Table node sub type (Temporal/GraphEdge/GraphNode)
+		 */
+		export enum TableSubType {
```

Comment:
This seems like something that we would want at a higher level - for example OE cares about the type of table.

Reply:
OE already has those changes in nodeSubType, so we don't need it

Reply:
What do you mean "has those changes"? I'm saying this enum should be something that is defined outside of the context of the table designer (but then table designer code references). It doesn't make sense to scope this enum down to the table designer level.

Reply:
OE has a node sub type for every node type. For table designer, we just need the node sub type for a table node type. We don't this subset of information in OE.

Reply:
So I'm not seeing a definition for the nodesubtype (it's just a string in the definition)

So I'd say this is an opportunity to more cleanly define that value. 

1. Define this TableSubType at the top level
2. Create a new type `export type NodeSubType = TableSubType;` - and then later on we can add more enums for the other different subtypes as needed
3. Update the nodeSubType type to be `nodeSubType?: string | NodeSubType | undefined`

Reply:
1. Lets rename this to TableIcon to reflect its real usage.
2. missing the default table icon here.